test/Neuron/Zettelkasten/TagSpec.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE ViewPatterns #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+module Neuron.Zettelkasten.TagSpec
+  ( spec,
+  )
+where
+
+import qualified Neuron.Zettelkasten.Tag as Z
+import Relude
+import Test.Hspec
+
+shouldMatch :: Z.TagPattern -> Z.Tag -> Expectation
+shouldMatch pat tag
+  | Z.tagMatch pat tag = pure ()
+  | otherwise = expectationFailure $ toString $ Z.tagPatternToText pat <> " was expected to match " <> Z.tagToText tag <> " but didn't"
+
+shouldNotMatch :: Z.TagPattern -> Z.Tag -> Expectation
+shouldNotMatch pat tag
+  | Z.tagMatch pat tag = expectationFailure $ toString $ Z.tagPatternToText pat <> " wasn't expected to match tag " <> Z.tagToText tag <> " but it did"
+  | otherwise = pure ()
```

Comment:
These functions should be moved to the end of the file (`spec` should appear at the top).